llm-models/llamav2/llamav2-7b/02_mlflow_logging_inference.py

```diff
@@ -0,0 +1,263 @@
+# MAGIC %md
+# MAGIC ## Register the model to Unity Catalog
+# MAGIC  By default, MLflow registers models in the Databricks workspace model registry. To register models in Unity Catalog instead, we follow the [documentation](https://docs.databricks.com/machine-learning/manage-model-lifecycle/index.html) and set the registry server as Databricks Unity Catalog.
+# MAGIC
+# MAGIC  In order to register a model in Unity Catalog, there are [several requirements](https://docs.databricks.com/machine-learning/manage-model-lifecycle/index.html#requirements), such as Unity Catalog must be enabled in your workspace.
+# MAGIC
+
+# COMMAND ----------
+
+# Configure MLflow Python client to register model in Unity Catalog
+import mlflow
+mlflow.set_registry_uri("databricks-uc")
+
+# COMMAND ----------
+
+# Register model to Unity Catalog
+# This may take 2.2 minutes to complete
+
+registered_name = "models.default.llamav2_7b_model" # Note that the UC model name follows the pattern <catalog_name>.<schema_name>.<model_name>, corresponding to the catalog, schema, and registered model name
+
+
+result = mlflow.register_model(
+    "runs:/"+run.info.run_id+"/model",
+    registered_name,
+)
+
+# COMMAND ----------
+
+from mlflow import MlflowClient
+client = MlflowClient()
+client.set_registered_model_alias(registered_name, "Champion", 1)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Load the model from Unity Catalog
+
+# COMMAND ----------
+
+import mlflow
+import pandas as pd
+
+registered_name = "models.default.llama2_7b_model"
+loaded_model = mlflow.pyfunc.load_model(f"models:/{registered_name}@Champion")
+
+# Make a prediction using the loaded model
+loaded_model.predict(
+    {
+        "prompt": ["What is ML?", "What is large language model?"],
+        "temperature": [0.1, 0.5],
+        "max_new_tokens": [100, 100],
+    }
+)
+
+# COMMAND ----------
+
+# MAGIC %md
+# MAGIC ## Create Model Serving Endpoint
+# MAGIC Once the model is registered, we can use API to create a Databricks GPU Model Serving Endpoint that serves the `LLaMAV2-7b` model.
+# MAGIC
+
+# COMMAND ----------
+
+# Provide a name to the serving endpoint
+endpoint_name = 'llama2-7b-example'
+
+# COMMAND ----------
+
+databricks_url = dbutils.notebook.entry_point.getDbutils().notebook().getContext().apiUrl().getOrElse(None)
+token = dbutils.notebook.entry_point.getDbutils().notebook().getContext().apiToken().getOrElse(None)
+
+# COMMAND ----------
+
+import requests
+import json
+
+deploy_headers = {'Authorization': f'Bearer {token}', 'Content-Type': 'application/json'}
+deploy_url = f'{databricks_url}/api/2.0/serving-endpoints'
+
+model_version = result  # the returned result of mlflow.register_model
+endpoint_config = {
+  "name": endpoint_name,
+  "config": {
+    "served_models": [{
+      "name": f'{model_version.name.replace(".", "_")}_{model_version.version}',
+      "model_name": model_version.name,
+      "model_version": model_version.version,
+      "workload_type": "GPU_SMALL",
```

Comment:
Is `GPU_SMALL` a new feature?
From the external doc (https://docs.google.com/document/d/1SlNPRnQnnB_Qs9PhcW22ZVVJR0QIalcBY0mRdIdIrAw/edit) it says
> `workload_type` must be one of {“CPU”, “GPU_MEDIUM”}